analyzer/analyzer.go

```diff
@@ -16,23 +16,41 @@ import (
 	"golang.org/x/tools/go/analysis"
 )
 
+type optionInt struct {
+	enabled bool
+	intConv bool
+}
+
+type optionErr struct {
+	enabled  bool
+	errError bool
+	errorf   bool
+}
+
+type optionStr struct {
+	enabled   bool
+	sprintf1  bool
+	strconcat bool
+}
+
 type perfSprint struct {
-	intConv    bool
-	errError   bool
-	errorf     bool
-	sprintf1   bool
+	intFormat optionInt
+	errFormat optionErr
+	strFormat optionStr
+
+	boolFormat bool
+	hexFormat  bool
 	fiximports bool
-	strconcat  bool
 }
 
 func newPerfSprint() *perfSprint {
 	return &perfSprint{
-		intConv:    true,
-		errError:   false,
-		errorf:     true,
-		sprintf1:   true,
+		intFormat:  optionInt{enabled: true, intConv: true},
+		errFormat:  optionErr{enabled: true, errError: false, errorf: true},
+		strFormat:  optionStr{enabled: true, sprintf1: true, strconcat: true},
```

Comment:
Also, now the option is [inverted](https://github.com/catenacyber/perfsprint/pull/41#discussion_r1935362061), it's simplify code here

```suggestion
		intFormat:  optionInt{intConv: true},
		errFormat:  optionErr{errError: false, errorf: true},
		strFormat:  optionStr{sprintf1: true, strconcat: true},
```
Because the default value of deactivated, doesn't need to be explicit herr

Here I kept `errError: false` for clarity purpose

Reply:
ok